tensorflow/core/graph/graph.cc

```diff
@@ -453,6 +453,20 @@ const Edge* Graph::AddControlEdge(Node* source, Node* dest,
   return AddEdge(source, kControlSlot, dest, kControlSlot);
 }
 
+void Graph::RemoveControlEdge(const Edge* e, bool update_node_def) {
```

Comment:
Do you have need for update_node_def=false? If not, let's just remove it. The contract with AddControLEdge() is that it will make sure the NodeDef is in sync with the added control edge (unless allow_duplicates=true, which is only in one weird specific case that we would ideally change). So we can assume that we should continue to keep the NodeDef in sync here.

Reply:
Yeah I only added that option in order to be consistent with your previous change because I was unsure of where else it might be necessary to not keep the NodeDef in sync. I'll remove that.